File: Sorts/insertion_sort/sort.py
def insertion_sort(arr , value):
  left = 0 
  right = len(arr)-1
  mid = 0
  while left <=right:
    mid = (left + right)//2
    if (value < arr[mid]):
      right = mid - 1
    elif (value > arr[mid]):
      left = mid +1 
    else:
      return mid 
  return -1

vetor = [ 1 , 2 , 3 , 4 , 5 , 6 , 7 ]
aux = insertion_sort(vetor , 7)
print(aux)

File: Sorts/insertion_sort/sort.cpp
#include <iostream>
using namespace std;

void insertionSort( int arr[] , int n){
  
  for(int j=1 ; j<n; j++){
    int chave = arr[j];
    int i = j - 1;
    while( i >= 0 && arr[i] > chave){
      arr[i+1]=arr[i];
      i--;
    }
    arr[i+1] = chave;

  }
}
int main() {
  int vetor[] {5 , 3 , 2 , 4};
  
  insertionSort(vetor , 4);
  for(int aux = 0 ; aux < 4 ; aux++){
    cout << vetor[aux] << endl;
  }
}